# **PacCommerce - Customer Membership**
---

## **Case Study**

To check whether the program is working properly, we need to test it on a few test cases below.

First, we need to import the Membership class from the main.py file

In [1]:
from main import Membership

### **Case 1**

- Cahya wants to check all membership tier and the benefits available in PacCommerce.

- **input** : -

- **output** : List of available membership tiers and their corresponding benefits in PacCommerce.

In [4]:
# Create a new user named Cahya
user1 = Membership('Cahya') 

In [3]:
# Checking all the membership and its benefit available
# using .show_benefit() funciton
user1.show_benefit()

| Membership   | Discount   | Another Benefit                                             |
|--------------|------------|-------------------------------------------------------------|
| Platinum     | 15%        | Benefit Silver + Gold + Voucher Liburan + Cashback max. 30% |
| Gold         | 10%        | Benefit Silver + Voucher Ojek Online                        |
| Silver       | 8%         | Voucher Makanan                                             |


### **Case 2**

- Shandy wants to know the requirements for monthly expenses and monthly income needed to obtain a predetermined membership.

- **input** : -

- **output** : Membership tier, monthly expense and income requirement for each tier.

In [5]:
# Create a new user named Shandy
user2 = Membership('Shandy')

In [6]:
# Showing all membership tier and its monthly income and expense requirement
# using .show_requirements()
user2.show_requirements()

|   Platinum |       Gold |    Silver |
|------------|------------|-----------|
| 15,000,000 | 10,000,000 | 7,000,000 |
|  8,000,000 |  6,000,000 | 5,000,000 |


### **Case 3**

- User Cahya wants to know which membership category they belong to based on his monthly expenses and monthly income. 
    - monthly_income = IDR 12.000.000,00 
    - monthly_expense = IDR 9.000.000,00

- **input** : 
    - monthly_income
    - monthly_expense

- **output** :
    - membership_tier

In [7]:
# first, let's set some value for Cahya monthly income and expense
user1 =  Membership('Cahya', monthly_income=12_000_000,
                    monthly_expense=9_000_000)

In [8]:
# Now let's predict which tier Cahya belong to using .predict_membership()
user1.predict_membership()

Hi Cahya
Based on your income and expense, you're eligible for Platinum Membership


### **Case 4**

- Now Cahya want to buy a lot of things in PacCommerce.
    
- List of Cahya PacCommerce cart :
    - 1 kg of Beef for IDR 150.000,00
    - 10 kg of Rice for IDR 200.000,00
    - 2 kg of Red Chili for IDR 400.000,00

In [9]:
# let's store Cahya shopping cart in a list
cahya_shopping_cart = [150_000, 200_000, 400_000]

In [10]:
# Now let's check the final price using calculate_price()
user1.calculate_price(cahya_shopping_cart)

Rp. 637.500,00


In [11]:
# let's compare if Cahya is paying the original price
sum(cahya_shopping_cart)

750000

As you can see, Cahya is getting 15% Discount because he have Platinum Tier Memberhip.